fix: [SCRUM-191] Change hardcoded IDs to use globalData

BaoLam-hue opened the pull request:
Change listening test navigation and name.
Fix listening-test UI.
Remove all remaining unused matching code.
Create useGlobalData hook for fetching globalData key from local storage.

BaoLam-hue committed:
Fix: [SCRUM-191] Add sessionId due to API changes

BaoLam-hue committed:
change sessionId temporary

BaoLam-hue committed:
Merge branch 'dev-v2' of https://github.com/khoanguyen-st/greenPREP into refactor/SCRUM-191

BaoLam-hue committed:
Merge branch 'dev-v2' of https://github.com/khoanguyen-st/greenPREP into refactor/SCRUM-191

BaoLam-hue committed:
Merge branch 'dev-v2' of https://github.com/khoanguyen-st/greenPREP into refactor/SCRUM-191

BaoLam-hue committed:
fix: [SCRUM-191] Change hardcoded IDs to use globalData

BaoLam-hue committed:
Merge branch 'dev-v2' of https://github.com/khoanguyen-st/greenPREP into refactor/SCRUM-191

khoanguyen-st commented on src/features/listening/api/listeningAPI.js:
```diff
@@ -1,14 +1,34 @@
-  })
-  return response.data
+
+    const response = await axios.get(`${API_BASE_URL}/topics/${topicId}`, {
```
Blocking: Using `axiosinstance`

BaoLam-hue replied:
Fixed

khoanguyen-st commented on src/pages/listening/listening-test.jsx:
```diff
@@ -27,15 +25,54 @@ const ListeningTest = () => {
+      setErrorMessage('Failed to load test data. Please refresh the page or contact support.')
+      setShowErrorModal(true)
+      return null
+    }
```
Blocking: Please define a custom hook for this method

BaoLam-hue replied:
Defined useGlobalData hook

BaoLam-hue committed:
Merge branch 'dev-v2' of https://github.com/khoanguyen-st/greenPREP into refactor/SCRUM-191

BaoLam-hue committed:
Change listening test navigation and name.
Fix listening-test UI.
Remove all remaining unused matching code.
Create useGlobalData hook for fetching globalData key from local storage.

khoanguyen-st commented on src/features/listening/api/index.js:
```diff
@@ -0,0 +1,37 @@
+      throw new Error('Missing topicId in localStorage.globalData')
+    }
+
+    const response = await axios.get(`/topics/${topicId}`, {
```
Blocking: using axiosinstance

BaoLam-hue committed:
use axiosInstance
